.github/canary-release.js

```diff
@@ -0,0 +1,28 @@
+import fs from 'fs';
+import { exec } from 'child_process';
+
+const pkgJsonPaths = ['cli/package.json'];
+
+try {
+  exec('git rev-parse --short HEAD', (err, stdout) => {
+    if (err) {
+      console.log(err);
+      process.exit(1);
+    }
+    const commitHash = stdout.trim();
+
+    for (const pkgJsonPath of pkgJsonPaths) {
+      const pkg = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf-8'));
+      const oldVersion = pkg.version;
+      const [major, minor, patch] = oldVersion.split('.').map(Number);
+      const newVersion = `${major}.${minor}.${patch + 1}-canary.${commitHash}`;
+      pkg.version = newVersion;
+      const content = JSON.stringify(pkg, null, '\t') + '\n';
```

Comment:
Never actually seen a `\t` used here, typically it's a `2`. I guess this depends on if the project uses tabs or spaces? Haven't thought about that in a long time :)

```suggestion
      const content = JSON.stringify(pkg, null, 2) + '\n';
```